fip-x.md

```diff
@@ -0,0 +1,115 @@
+
+Proposed new actions:
+|Action|Description|
+|---|---|
+|list|User puts domain up for sale|
+|cancel_listing|Seller of domain decides not to sell and cancels their listing|
+|buy_listing|User buys a domain that is up for sale|
+
+# Terminology
+* **Listing** - a domain that is up for sale
+* **Marketplace** - the frontend website
+* **Seller** - the seller of a domain
+* **Buyer** - the buyer of a domain
+
+# Motivation
+There does not currently exist a platform for buying, selling, and trading FIO Domains.  As the popularity of FIO grows, the likelihood of a desired domain being available decreases.  It is also common for individuals to purchase names of popular name brands in order to later sell the domain to the company who owns the brand (name squatting).  Without a marketplace for domains to be sold, these transactions become difficult.  By providing ease of use solutions to name and address transactions, a marketplace also increases the revenue flow of block producers by increasing the amount of fees generated by each transaction.
+
+# Specification
+
+## Core Concept
+Users will interact with the frontend website to view and place domains up for sale.  The frontend will interact with the FIO Escrow Smart Contract to facilitate transactions between users who buy and sell domains.  The diagram below displays how the user, the website, the smart contract and the account that owns the smart contract all work together. 
+
+![Image of UML Diagram](https://github.com/iwill610/FDM_UMLsd/blob/main/FIO%20Domain%20Marketplace.png)
+
+This diagram reads left to right and top to bottom.  There are three different scenarios outlined and they are distinguished by the color of the arrows.  The first scenario, in red, starts will the user using the website to place a domain up for sale for a specified price.  This user is denoted as "seller."  The website then communicates that the smart contract needs to use the "list" action, which will transfer ownership of the address to the contract account and place that address for sale on the website.  
+
+The second scenario, in blue, starts will the seller cancelling their listing on the website.  The website then communicates with the smart contract in order to transfer ownership of the domain back to the seller.  
+
+The last scenario, in green, starts with a different user buying a listing on the website.  This second user is denoted as "buyer."  The website then communicates with the smart contract in order to transfer ownership of the address to the buyer and transfer the funds used to buy the domain to the seller. 
+
+## New actions
+### List
+This action will post the domain up for sale on the marketplace for a specified price. 
+#### New action: *list*
```

Comment:
Should this be more specific to marketplace and/or domain, e.g. listdomain, cxlistdomain, buydomain
Please note action name can be max 12 characters long.

Reply:
I think the action names you provided make more sense and are more cohesive.  It's also good to make this distinction early because we have plans on implementing functionality for addresses in the future. Thanks